### ISSUE TITLE
The presence of an HTML-based figure caption prevents image previews in the editor

### ISSUE THREAD

**Issue by michaellenaghan** [typlog/issues#28]
When an HTML-based figure caption follows an image URL in the editor, the editor doesn't show an image preview:

<img width="850" alt="Screenshot 2022-11-29 at 6 59 00 PM" src="https://user-images.githubusercontent.com/176481/204675039-8611d314-8bb9-4d55-9d1f-9979de6acdc8.png">

Adding a blank line makes the image preview appear:

<img width="945" alt="Screenshot 2022-11-29 at 6 59 15 PM" src="https://user-images.githubusercontent.com/176481/204675061-9d30e258-e748-4576-8008-52bdd0dffe04.png">

but it also makes the HTML a separate paragraph rather than a caption.

**Comment by lepture** [comment]
@michaellenaghan yes. It is because markdown has no block level image syntax. I just released mistune 3.0.0rc4, in this version, there is a `figure` directive:

````
```{figure} src
:alt: alt text

this can be **strong** caption
```
````

**Comment by michaellenaghan** [comment]
Understood. (I assume that means captions for galleries are unsupported.)
